code/lib/node-logger/src/index.ts

```diff
@@ -25,13 +25,28 @@ export const logger = {
   plain: (message: string): void => console.log(message),
   line: (count = 1): void => console.log(`${Array(count - 1).fill('\n')}`),
   warn: (message: string): void => npmLog.warn('', message),
-  // npmLog supports anything we log, it will just stringify it
-  error: (message: unknown): void => npmLog.error('', message as string),
   trace: ({ message, time }: { message: string; time: [number, number] }): void =>
     npmLog.info('', `${message} (${colors.purple(prettyTime(time))})`),
   setLevel: (level = 'info'): void => {
     npmLog.level = level;
   },
+  error: (message: Error | string): void => {
+    if (npmLog.levels[npmLog.level] < npmLog.levels.error) {
+      let msg: string;
+
+      if (message instanceof Error && message.stack) {
+        msg = message.stack.toString();
+      } else {
+        msg = message.toString();
+      }
+
+      console.log(
+        msg
+          .replaceAll(process.cwd(), '.')
+          .replace(message.toString(), chalk.red(message.toString()))
```

Comment:
You might want to swap them around, else message.toString() which could contain a file path, will not be the same as `msg` after the path is changed

```suggestion
          .replace(message.toString(), chalk.red(message.toString()))
          .replaceAll(process.cwd(), '.')
```

Reply:
FYI this is what it looks like when the error message already contains ansi characters from chalk

<img width="642" alt="image" src="https://github.com/storybookjs/storybook/assets/1671563/2f65d635-31b1-45a6-9611-d0f75a35bae4">